airflow/models/baseoperator.py

```diff
@@ -380,7 +387,15 @@ def __init__(
                 stacklevel=3
             )
         validate_key(task_id)
-        self.task_id = task_id
+        self._task_id = task_id
```

Comment:
I am a bit concerned with us changing the semantic of `self.task_id` here for a UI feature. There are airflow code written with the assumption this is the id that can be used for querying in database.

If the task_group prefixed "task_id" is only intended for visualization, would it make more sense to create a new attribute for it instead of overriding the existing one?

Reply:
You are right. i'm reverting this back to `self.task_id`.

That said, I don't think we should allow people to change `task_id` of a task once it's set. That causes the task_dict to have a wrong key and will certainly cause bugs. So maybe it's better to have `task_id` as a readonly propertly. That is out of the scope of this PR. So i'll leave things in the original state for now.